### PR TITLE
core 3171 js fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
With the upcoming gregor avdl stuff we'll need to be import aware and namespace aware. In the shortterm this just makes the build continue to work.

A side effect of the v2 json generation is we don't pull in imports into each file so things like ClientType exist in Common instead of being copied all over our keybase-v1 file. This required pointing to the single source of truth which caused the file count to go up. 

But... now things are flow typed better and we lost a bunch of lines of bs code

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -7,6 +7,8 @@
 /react-native/shared
 /shared/libs/immutable-interface.js
 /shared/libs/flow-interface.js
+/shared/constants/types/keybase-v1.js
```
Comment (chrisnojima) [inline .eslintignore]: we don't necessarily control the data so lets not break ourselves if this doesn't pass link

File: go/protocol/prove_common.go [hunk 2]
```diff
@@ -82,16 +82,16 @@ const (
 	ProofType_ROOTER           ProofType = 100001
 )
 
-type Prove_commonInterface interface {
```
Comment (chrisnojima) [inline go/protocol/prove_common.go]: @maxtaco so you expect this won't break anything right

File: protocol/avdl/prove_common.avdl [hunk 3]
```diff
@@ -1,5 +1,5 @@
 @namespace("keybase.1")
-protocol prove_common {
+protocol proveCommon {
```
Comment (chrisnojima) [inline protocol/avdl/prove_common.avdl]: this is the only one that isn't camelcase, @maxtaco said i should make it conform

File: protocol/bin/flow.js [hunk 4]
```diff
@@ -17,8 +17,8 @@ fs
 
 let incomingMaps = {}
 
-function jsonOnly (file) {
-  return !!file.match(/.*\.json$/)
+function jsonOnlySkipGregor (file) {
+  return file !== 'gregor.json' && !!file.match(/.*\.json$/)
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: we ignore the gregor stuff for now

File: protocol/bin/flow.js [hunk 5]
```diff
@@ -50,8 +50,11 @@ function analyze (json) {
 }
 
 function figureType (type) {
+  if (!type) {
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: the v2 flag brings in imports (cool) but now acts slightly differently (boo). this makes our script not have to deal with this too much

File: protocol/bin/js.js [hunk 6]
```diff
@@ -38,6 +38,11 @@ function addEnums (prot, json) {
 function write () {
   var stream = fs.createWriteStream('js/keybase-v1.js')
   stream.once('open', function (fd) {
+    stream.write(`/* @flow */
```
Comment (chrisnojima) [inline protocol/bin/js.js]: keybase-v1.js is now flow typed

File: protocol/js/keybase-v1.js [hunk 7]
```diff
@@ -1,37 +1,16 @@
-export const account = {
```
Comment (chrisnojima) [inline protocol/js/keybase-v1.js]: a side effect of the new json creator is we don't have dupes all over the place

File: shared/actions/profile.js [hunk 8]
```diff
@@ -2,10 +2,9 @@
 import * as Constants from '../constants/profile'
 import {routeAppend} from './router'
 import engine from '../engine'
-import {identify} from '../constants/types/keybase-v1'
+import {identifyCommon, Common, proveCommon} from '../constants/types/keybase-v1'
```
Comment (chrisnojima) [inline shared/actions/profile.js]: pull in the types we need. these moved around slightly

File: shared/util/kbfs-notifications.js [hunk 9]
```diff
@@ -52,29 +52,35 @@ export function decodeKBFSError (user: string, notification: FSNotification): De
...
-      title: 'Keybase: KBFS error',
-      body: `${notification.status}`
-    })
-  }
+  return ({
+    title: 'Keybase: KBFS error',
+    body: `${notification.status}`
+  })
+
+  // This code came from the kbfs team but this isn't plumbed through the protocol. Leaving this for now
```
Comment (chrisnojima) [inline shared/util/kbfs-notifications.js]: feature doesn't exist in the protocol so just commenting this out

**Comment by chrisnojima** [comment]
@keybase/react-hackers @maxtaco

File: protocol/bin/flow.js [hunk 10]
```diff
@@ -72,6 +75,7 @@ function capitalize (s) {
 }
 
 function analyzeMessages (json) {
+  console.log(json)
```
Comment (chromakode) [inline protocol/bin/flow.js]: Was this log statement intended to be checked in?

File: go/protocol/prove_common.go (hunk 2 above)
Comment (maxtaco) [inline go/protocol/prove_common.go]: that is my expectation!

File: protocol/avdl/prove_common.avdl (hunk 3 above)
Comment (maxtaco) [inline protocol/avdl/prove_common.avdl]: 👍

File: protocol/bin/flow.js (hunk 4 above)
Comment (maxtaco) [inline protocol/bin/flow.js]: i'll add a few others as i go, but cool

**Comment by maxtaco** [comment]
looks good to me

**Comment by maxtaco** [comment]
but we should also get a look from @keybase/react-hackers

**Comment by chrisnojima** [comment]
@chromakode confused about your comment. i removed that 3 hours ago but you saw it and commented 4 minutes ago and it immediately was marked as out of date? maybe your browser didn't refresh?

**Comment by chrisnojima** [comment]
or github lag?

**Comment by chromakode** [comment]
@chrisnojima it's added in one commit and removed in the next. Hadn't gotten to the second commit yet

**Comment by chrisnojima** [comment]
k, kinda confusing

**Comment by chromakode** [comment]
:+1: Seems fine to me but I don't have many insights into potential pitfalls of these changes.

**Comment by maxtaco** [comment]
Big thanks @chrisnojima !!!

**Pull request merged**